lesson_github_collaboration.Rmd

```diff
@@ -0,0 +1,84 @@
+---
+title: "Github Collaboration: Branching & Pull-Requests"
+---
+
+<br>
+
+This is a quick tutorial in how to use *branches* in github, in addition to *Pull requests*, which is a way to create new content, review it, and then merge it back to your master content (the `master branch`). This is something that might take a bit of practice initially, but the process is and important "best practice" for collaboration and version control.
+
+Ideally you only want to do one thing per branch that you work on...whether that's fixing some code in a specific script, or adding new content (but only a single piece), the simpler you can keep what you add to a new branch, the easier it will be to track, and merge back into the master branch.
+
+## Create a New Branch
+
+Make sure you're on the `master` branch, and then either create a new branch via github, or in the command line use:
+
+ - `git checkout -b YOUR-NEW-BRANCH`
+
+### Make some content
+
+Once that's done, you can make your changes and update your content as needed. Once done, add one or more files (typically focus on adding one Rmd or lesson per branch). Perhaps most important is only adding the files you are working on or that you have revised (so **don't** use `git add . -A`). Rather use:
+
+ - `git add lesson_on_something_cool.Rmd`
+ - `git add data/data_for_something_cool.rda`
+ - `git add img/some_imgs_for_something_cool.png`
+
+### Commit Changes
+
+Now comes time to commit your changes. Add a message that is detailed but not too long.
+
+ - `git commit -m "add something_cool lesson"`
+
+### Push Changes
+
+When you want to push, make sure you are on the branch with your content, add the branch to the upstream `master`:
+
```

Comment:
Interesting. I don't do this step to add the branch to the upstream master. Is that bad?

After I commit, I go straight to `git push origin YOUR-BRANCH-NAME`

Reply:
Good point... I think I may be using a slightly different option (same thing, just added step which explicitly adds the branch...but isn't necessary). I think your method is good...and might be a bit simpler. When you git push origin I think it adds it automagically to the master and pushes changes all at once.